markets/retrievaladapter/client.go

```diff
@@ -9,23 +9,54 @@ import (
 	"github.com/filecoin-project/go-state-types/abi"
 	"github.com/ipfs/go-cid"
 	"github.com/multiformats/go-multiaddr"
+	mh "github.com/multiformats/go-multihash"
+	"golang.org/x/xerrors"
 
+	"github.com/filecoin-project/lotus/api"
 	"github.com/filecoin-project/lotus/chain/actors/builtin/paych"
 	"github.com/filecoin-project/lotus/chain/types"
 	"github.com/filecoin-project/lotus/node/impl/full"
 	payapi "github.com/filecoin-project/lotus/node/impl/paych"
 )
 
+func mkPaychReusedCid(addr address.Address) cid.Cid {
```

Comment:
I feel somewhat strongly that rather than do all this hackery with addresses and cids, we should properly handle cid.Undef over in markets. To that end, I've written a PR to do so -- https://github.com/filecoin-project/go-fil-markets/pull/677